### PR TITLE
Fixed drawBarChart in case maxVal=1 (added the line x=1 to vertical g…

### PR DESCRIPTION
**Pull request by ikarus512**
Added the line x=1 to vertical grid lines (x=0, x=10, x=100, ...).
Fixed drawBarChart in case maxVal=1: now maximum x on chart is 1 (it was 10).

### CODE REVIEW

File: src/tFPDF/GraphicReportPDF.php [hunk 1]
```diff
@@ -606,21 +606,21 @@ private function setLegends($data) {
...
+        for ($i = 0; $i <= $maxGrid+1; $i++ ) {
             if ($i === 0) {
                 $this->setDrawColor($red, $green, $blue);
             } else {
                 $this->setDrawColor(200,200,200);
             }
             $xpos = $XDiag + $stepValue * $i;
             $this->Line($xpos, $YDiag - (($i === 0) ? 5 : 0), $xpos, $YDiag + $hDiag);
-            $val = $i === 0 ? 0 : pow(10, $i);
+            $val = $i === 0 ? 0 : pow(10, $i-1);
```
Comment (bigxoma) [inline src/tFPDF/GraphicReportPDF.php]: add spaces please

File: src/tFPDF/GraphicReportPDF.php [hunk 2]
```diff
@@ -606,21 +606,21 @@ private function setLegends($data) {
     private function barChartDrawScales($XDiag, $YDiag, $hDiag, $margin, $maxGrid, $stepValue) {
         list($red, $green, $blue) = self::COLOR_BLACK;
         $this->SetLineWidth(0.1);
-        for ($i = 0; $i <= $maxGrid; $i++ ) {
+        for ($i = 0; $i <= $maxGrid+1; $i++ ) {
```
Comment (bigxoma) [inline src/tFPDF/GraphicReportPDF.php]: add spaces please

File: src/tFPDF/GraphicReportPDF.php [hunk 3]
```diff
@@ -606,21 +606,21 @@ private function setLegends($data) {
...
             $this->Line($xpos, $YDiag - (($i === 0) ? 5 : 0), $xpos, $YDiag + $hDiag);
-            $val = $i === 0 ? 0 : pow(10, $i);
+            $val = $i === 0 ? 0 : pow(10, $i-1);
             $ypos = $YDiag + $hDiag - $margin + 5;
             $this->Text($xpos, $ypos, $val);
         }
         $this->SetLineWidth(0.2);
         $this->setDrawColor($red, $green, $blue);
-        $this->Line($XDiag, $YDiag + $hDiag - $margin + 2, $XDiag + $stepValue * $maxGrid + 5, $YDiag + $hDiag - $margin + 2);
+        $this->Line($XDiag, $YDiag + $hDiag - $margin + 2, $XDiag + $stepValue * ($maxGrid+1) + 5, $YDiag + $hDiag - $margin + 2);
```
Comment (bigxoma) [inline src/tFPDF/GraphicReportPDF.php]: add spaces please

File: src/tFPDF/GraphicReportPDF.php [hunk 4]
```diff
@@ -968,7 +965,7 @@ private function calculateDataForBarDiagram($h, $barChartData, $maxVal, $nbDiv):
         $valIndRepere = ceil($maxVal / $nbDiv);
         $maxVal = $valIndRepere * $nbDiv;
         $lRepere = floor($lDiag / $nbDiv);
-        $pxDecade = floor($lDiag / $maxGrid);
+        $pxDecade = floor($lDiag / ($maxGrid+1));
```
Comment (bigxoma) [inline src/tFPDF/GraphicReportPDF.php]: add spaces please

File: src/tFPDF/GraphicReportPDF.php (hunk 1 above)
Comment (ikarus512) [reply src/tFPDF/GraphicReportPDF.php]: Done

File: src/tFPDF/GraphicReportPDF.php (hunk 2 above)
Comment (ikarus512) [reply src/tFPDF/GraphicReportPDF.php]: Done

File: src/tFPDF/GraphicReportPDF.php (hunk 3 above)
Comment (ikarus512) [reply src/tFPDF/GraphicReportPDF.php]: Done

File: src/tFPDF/GraphicReportPDF.php (hunk 4 above)
Comment (ikarus512) [reply src/tFPDF/GraphicReportPDF.php]: Done

**Review by bigxoma — APPROVED** [review]

**Pull request merged**